cmd/xl-v1-healing-common.go

```diff
@@ -102,18 +96,10 @@ func listOnlineDisks(disks []StorageAPI, partsMetadata []xlMetaV1, errs []error)
 	return onlineDisks, modTime
 }
 
-// Return disks with the outdated or missing object.
-func outDatedDisks(disks []StorageAPI, partsMetadata []xlMetaV1, errs []error) (outDatedDisks []StorageAPI) {
+// outDatedDisks - return disks which don't have the latest object (i.e xl.json).
+func outDatedDisks(disks, latestDisks []StorageAPI, partsMetadata []xlMetaV1, errs []error) (outDatedDisks []StorageAPI) {
 	outDatedDisks = make([]StorageAPI, len(disks))
-	latestDisks, _ := listOnlineDisks(disks, partsMetadata, errs)
 	for index, disk := range latestDisks {
```

Comment:
errors other than errFileNotFound should not be considered as outdatedDisk, because if the error was errDiskNotFound erasureHealFile will try to write on it and fail - so healing will fail.